sdk/ml/azure-ai-ml/tests/pipeline_job/_util.py

```diff
@@ -56,3 +56,28 @@
     lambda params: Path(params[0]).name,
     DATABINDING_EXPRESSION_TEST_CASES,
 )))
+
+
+ERROR_MESSAGE_INVALID_RESOURCE_WITHOUT_COMPUTE = (
+    "Invalid resource is set, please set serverless compute or pipeline default compute"
```

Comment:
how do we determine what's Invalid resource?

Reply:
May refer to above comment, this corresponds to combination "null default compute of pipeline + resources compute of step", where the resources can be regarded as invalid to some degree (or say in pipelines). Not sure "invalid resource" is suitable, I think the latter sentence is better but not suitable to be first part of error message.

Reply:
Have removed this error message, just "No compute is set" for this case.